ocaml/testsuite/tests/typing-layouts-products/unboxed_tuples.ml

```diff
@@ -0,0 +1,379 @@
+(* TEST
+ reference = "${test_source_directory}/unboxed_tuples.reference";
+ include stdlib_upstream_compatible;
+ flambda2;
```

Comment:
We need to test unboxed tuples at `-Oclassic` and `-O3`.  Unsure how you want to set this up.  It is possible to select these using either attributes e.g. `[@@@ocaml.flambda_oclassic]` or using e.g. `ocamlopt_flags="-Oclassic"` in ocamltest.  These only work on a per-source-file basis though...

Reply:
The easiest is probably to take the product of the existing ocamltest stanzas with ones setting the level I guess.

Reply:
I have done this, but I was surprised to see this on `-Oclassic`:
```
> ========================================
> File "unboxed_tuples.ml", line 74, characters 36-63:
> 74 |   print_ints "Test 1, 14/3 inlined" ((div_mod [@inlined]) 14 3);
>                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Warning 55 [inlining-impossible]: Cannot inline: 
>   the optimizer did not know what function was being applied (is it marked [@inline never]?)
>   (the full inlining stack was: unboxed_tuples.ml:74,36--63)
> ========================================
```
This function is defined just above in the same file, and is not marked `@inline never`.  I've changed the `@inlined` to `@inlined hint` for now, but want to check if this is expected.

Reply:
Here is the relevant part of the `-drawflambda` output:
```
camlUnboxed_tuples__div_mod_5_5 =
  ((code_metadata
...
     (inlining_decision (Function_body_too_large 25))
```
The message in the warning is not very helpful, and I think it could be refined to mention the actual reason. I'll make a note to try to improve this.

Reply:
That all makes sense - thanks very much for the explanation!